### PR TITLE
generic PCR prep

### PR DESCRIPTION
**Pull request by eskila**
This PR actually includes updates to the PCR prep part 1 & 2.

**Changes**

- I created dropdowns for PCR plates so that the user can choose between all available PCR plates in the labware library
- I fixed a couple of bugs in the code, i.e. a bug that discards the first non-header line in the input CSV in part 1 of the protocol, and a bug that would repeat pipetting of mastermix into the destination plate after the task was already finished.
- I removed the unavailable p1000 multi pipette from part 2 of the protocol
- I cleaned up the code and created helper functions.

Crossing my fingers that I am doing this right :)

### CODE REVIEW

**Comment by eskila** [comment]
Should these be new protocols, or replace the old ones on the repo?

File: protocols/pcr_prep_part_1_gen2/pcr_prep-part1_gen2.ot2.apiv2.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    [left_pipette, right_pipette, master_mix_csv] = get_values(  # noqa: F821
+        "left_pipette", "right_pipette", "master_mix_csv")
+
+    if not left_pipette and not right_pipette:
+        raise Exception('You have to select at least 1 pipette.')
+
+    pipette_l = None
+    pipette_r = None
+
+    for pip, mount, slot in zip(
```
Comment (chazmanc) [inline protocols/pcr_prep_part_1_gen2/pcr_prep-part1_gen2.ot2.apiv2.py:19]: what is `slot` here?

File: protocols/pcr_prep_part_1_gen2/pcr_prep-part1_gen2.ot2.apiv2.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            tiprack = protocol_context.load_labware(rack, slot)
+            if mount == 'left':
+                pipette_l = protocol_context.load_instrument(
+                    pip, mount, tip_racks=[tiprack])
+            else:
+                pipette_r = protocol_context.load_instrument(
+                    pip, mount, tip_racks=[tiprack])
+
+    # labware setup
+    snaprack = protocol_context.load_labware(
```
Comment (chazmanc) [inline protocols/pcr_prep_part_1_gen2/pcr_prep-part1_gen2.ot2.apiv2.py:34]: Can labware be updated in fields.json here?

File: protocols/pcr_prep_part_2_gen2/pcr_prep-part2_gen2.ot2.apiv2.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        else:
+            return [pipette_r, pipette_l]
+
+
+def pipette_selector(small_pipette, large_pipette, volume):
+    """
+    This function will return the smallest volume pipette capable
+    of handling the given volume parameter.
+    """
+    if small_pipette and large_pipette:
```
Comment (chazmanc) [inline protocols/pcr_prep_part_2_gen2/pcr_prep-part2_gen2.ot2.apiv2.py:111]: Interesting idea, but can be accomplished with just a line of code (for example, `pipette = small_pipette if volume <= small_pipette.max_volume else large_pipette`)

**Review by chazmanc — COMMENTED** [review]
let's make the changes/have a discussion, then merge

**Comment by chazmanc** [comment]
> 

let's replace the old versions on the repo

**Review by chazmanc — APPROVED** [review]

File: protocols/pcr_prep_part_1_gen2/README.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+## Categories
+* PCR
+    * Mastermix Assembly
+
+## Description
+Part 1 of 2: Master Mix Assembly
+
+Links:
+* [Part 1: Master Mix Assembly](./pcr_prep_part_1)
+* [Part 2: Master Mix Distribution and DNA Transfer](./pcr_prep_part_2)
```
[the remark this answers is not in the document]
Comment (eskila) [reply protocols/pcr_prep_part_1_gen2/README.md]: That should be fixed now

File: protocols/pcr_prep_part_2_gen2/pcr_prep-part2_gen2.ot2.apiv2.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+     DNA_volume, DNA_well_plate, destination_well_plate] \
+      = get_values(  # noqa: F821
+        "number_of_samples", "left_pipette", 'right_pipette',
+        "mastermix_volume", "DNA_volume",
+        "DNA_well_plate", "destination_well_plate"
+     )
+
+    number_of_samples = int(number_of_samples)
+    mastermix_volume = float(mastermix_volume)
+    DNA_volume = float(DNA_volume)
```
[the remark this answers is not in the document]
Comment (eskila) [reply protocols/pcr_prep_part_2_gen2/pcr_prep-part2_gen2.ot2.apiv2.py]: removed casts

**Pull request merged**